plugins/inputs/ctrlx_datalayer/ctrlx_datalayer.go

```diff
@@ -0,0 +1,363 @@
+	"github.com/boschrexroth/ctrlx-datalayer-golang/pkg/token"
+	"github.com/google/uuid"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/influxdata/telegraf/metric"
+	httpconfig "github.com/influxdata/telegraf/plugins/common/http"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	jsonParser "github.com/influxdata/telegraf/plugins/parsers/json"
+)
+
+// This plugin is based on the official ctrlX CORE API. Documentation can be found in OpenAPI format at:
+// https://boschrexroth.github.io/rest-api-description/ctrlx-automation/ctrlx-core/
+// Used APIs are:
+// * ctrlX CORE - Authorization and Authentication API
+// * ctrlX CORE - Data Layer API
+//
+// All communication between the device and this input plugin is based
+// on https REST and HTML5 Server Sent Events (sse).
+
+//go:embed sample.conf
+var sampleConfig string
+
+// CtrlXDataLayer encapsulated the configuration as well as the state of this plugin.
+type CtrlXDataLayer struct {
+	Server   string `toml:"server"`
+	Username string `toml:"username"`
+	Password string `toml:"password"`
+
+	Log          telegraf.Logger `toml:"-"`
+	Subscription []Subscription
+
+	url    string
+	wg     sync.WaitGroup
+	cancel context.CancelFunc
+
+	acc          telegraf.Accumulator
+	connection   *http.Client
+	tokenManager token.TokenManager
+	httpconfig.HTTPClientConfig
+}
+
+// convertTimestamp2UnixTime converts the given Data Layer timestamp of the payload to UnixTime.
+func convertTimestamp2UnixTime(t int64) time.Time {
+	// 1 sec=1000 milisec=1000000 microsec=1000000000 nanosec.
+	// Convert from FILETIME (100-nanosecond intervals since January 1, 1601 UTC) to
+	// seconds and nanoseconds since January 1, 1970 UTC.
+	// Between Jan 1, 1601 and Jan 1, 1970 there are 11644473600 seconds.
+	return time.Unix(0, (t-116444736000000000)*100)
+}
+
+// createSubscription uses the official 'ctrlX Data Layer API' to create the sse subscription.
+func (c *CtrlXDataLayer) createSubscription(sub *Subscription) (string, error) {
+	sseURL := c.url + subscriptionPath
+
+	id := "telegraf_" + uuid.New().String()
+	request := sub.createRequest(id)
+	payload, err := json.Marshal(request)
+	if err != nil {
+		return "", fmt.Errorf("failed to create subscription %d payload, error: %w", sub.index, err)
+	}
+
+	requestBody := bytes.NewBuffer(payload)
+	req, err := http.NewRequest("POST", sseURL, requestBody)
+	if err != nil {
+		return "", fmt.Errorf("failed to create subscription %d request, error: %w", sub.index, err)
+	}
+
+	req.Header.Add("Authorization", c.tokenManager.Token.String())
+
+	resp, err := c.connection.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("failed to do request to create sse subscription %d, error: %w", sub.index, err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 && resp.StatusCode != 201 {
+		return "", fmt.Errorf("failed to create sse subscription %d, status: %q", sub.index, resp.Status)
```

Comment:
Does it need to be quoted?
```suggestion
		return "", fmt.Errorf("failed to create sse subscription %d, status: %s", sub.index, resp.Status)
```